### PR TITLE
Document extended resources and OIR deprecation.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
Implemented here: https://github.com/kubernetes/kubernetes/pull/48922

Already included in v1.8 release notes:
* https://github.com/kubernetes/features/blob/master/release-1.8/release_notes_draft.md#scheduling
* https://github.com/kubernetes/features/blob/master/release-1.8/release_notes_draft.md#behavioral-changes

/sig scheduling

cc @davidopp @vishh

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5399)

### CODE REVIEW

**Comment by balajismaniam** [comment]
For future reviewers: https://deploy-preview-5399--kubernetes-io-vnext-staging.netlify.com/docs/concepts/configuration/manage-compute-resources-container

File: docs/concepts/configuration/manage-compute-resources-container.md [hunk 1]
```diff
@@ -305,6 +305,15 @@ where `OOM` stands for Out Of Memory.
 
 ## Opaque integer resources (Alpha feature)
 
+{% include feature-state-deprecated.md %}
+
+**Note:** Opaque Integer Resources will be removed in version 1.9.
```
Comment (balajismaniam) [inline docs/concepts/configuration/manage-compute-resources-container.md]: Move this note below https://github.com/kubernetes/kubernetes.github.io/pull/5399/files?diff=unified#diff-4c9401ed6b037e1adb958cbce20630c7L321.

File: docs/concepts/configuration/manage-compute-resources-container.md [hunk 2]
```diff
@@ -395,6 +405,106 @@ spec:
...
+_invalid_ quantities are `0.5` and `1500m`.
+
+**Note:** Extended Resources replace [Opaque Integer
+Resources](#opaque-integer-resources-alpha-feature). Users can use any domain
+name prefix outside of the `kubernetes.io/` domain instead of the previous
+`pod.alpha.kubernetes.io/opaque-int-resource-` prefix.
+{: .note}
+
+There are two steps required to use Extended Resources. First, the
+cluster operator must advertise a per-node opaque resource on one or more
```
Comment (balajismaniam) [inline docs/concepts/configuration/manage-compute-resources-container.md]: s/opaque/extended or s/opaque//

File: docs/concepts/configuration/manage-compute-resources-container.md [hunk 3]
```diff
@@ -316,6 +325,7 @@ available amount is simultaneously allocated to Pods.
 **Note:** Opaque integer resources are Alpha in Kubernetes version 1.5.
```
Comment (ConnorDoyle) [inline docs/concepts/configuration/manage-compute-resources-container.md]: I think we can drop this note entirely now.

File: docs/concepts/configuration/manage-compute-resources-container.md [hunk 4]
```diff
@@ -395,6 +405,106 @@ spec:
...
+Host: k8s-master:8080
+
+[
+  {
+    "op": "add",
+    "path": "/status/capacity/example.com~1foo",
+    "value": "5"
+  }
+]
+```
```
[the remark this answers is not in the document]
Comment (ConnorDoyle) [reply docs/concepts/configuration/manage-compute-resources-container.md]: OK, should we remove the similar one from the OIR section above too?

File: docs/concepts/configuration/manage-compute-resources-container.md [hunk 5]
```diff
@@ -395,6 +405,106 @@ spec:
...
+http://k8s-master:8080/api/v1/nodes/k8s-node-1/status
+```
+
+**Note**: In the preceding request, `~1` is the encoding for the character `/`
+in the patch path. The operation path value in JSON-Patch is interpreted as a
+JSON-Pointer. For more details, see
+[IETF RFC 6901, section 3](https://tools.ietf.org/html/rfc6901#section-3).
+{: .note}
+
+To consume an Extended Resource in a Pod, include the resource name as a key
```
[the remark this answers is not in the document]
Comment (ConnorDoyle) [reply docs/concepts/configuration/manage-compute-resources-container.md]: I meant this sentence to be more general than the specific resource from the example. Did that not come across? This is the second of two steps required to use ERs. The previous section began with "To advertise a new Extended Resource..."

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tengqm) [reply docs/concepts/configuration/manage-compute-resources-container.md]: If there is no meat in it, better throw it away IMO.

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 3 above)
Comment (tengqm) [reply docs/concepts/configuration/manage-compute-resources-container.md]: agreed

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (tengqm) [reply docs/concepts/configuration/manage-compute-resources-container.md]: Alright, if it is referring to general cases, i.e. "an Extended Resource", it is fine.
I was reading it as "the Extended Resource".

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhangxiaoyu-zidif) [reply docs/concepts/configuration/manage-compute-resources-container.md]: I suggest to remain it. It's very important for usual users.
I can help user to config this kind of resource though shell.

File: docs/concepts/configuration/manage-compute-resources-container.md [hunk 6]
```diff
@@ -395,6 +405,106 @@ spec:
...
+    "value": "5"
+  }
+]
+```
+
+```shell
+curl --header "Content-Type: application/json-patch+json" \
+--request PATCH \
+--data '[{"op": "add", "path": "/status/capacity/example.com~1foo", "value": "5"}]' \
+http://k8s-master:8080/api/v1/nodes/k8s-node-1/status
```
[the remark this answers is not in the document]
Comment (zhangxiaoyu-zidif) [reply docs/concepts/configuration/manage-compute-resources-container.md]: @ConnorDoyle 
I think at least 50% common user do not know how to config OIR or Extended Resouce by curl quickly. So remaining it is very necessary. I meet many users ask how to config OIR, so I add it in OIR docs. =)

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tengqm) [reply docs/concepts/configuration/manage-compute-resources-container.md]: @zhangxiaoyu-zidif The snippet above is not a JSON or an YAML for you to create resource from Shell. It is an explanation of the HTTP request internals, and the information is duplicated with the command below.

[the remark this answers is not in the document]
Comment (zhangxiaoyu-zidif) [reply docs/concepts/configuration/manage-compute-resources-container.md]: Sorry to misunderstand your response. The command below is necessary. And the above one can be remove. It has no meaning to common users. But the below shell command is useful. =)

**Comment by ConnorDoyle** [comment]
cc @steveperry-53 (this should go in v1.8 docs)

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ConnorDoyle) [reply docs/concepts/configuration/manage-compute-resources-container.md]: My historical note is that I first added the HTTP request, then reviewers asked for a `curl` equivalent. Guess the rationale is that the K8s API speaks HTTP and not `curl`. But I'm apparently in the minority here. Will remove both of them.

File: docs/concepts/configuration/manage-compute-resources-container.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ConnorDoyle) [reply docs/concepts/configuration/manage-compute-resources-container.md]: @tengqm the URL shown is the correct way to patch the node status.

**Comment by ConnorDoyle** [comment]
@tengqm updated, PTAL

**Comment by zacharysarah** [comment]
@ConnorDoyle :wave: Do you want to wait for a LGTM from @tengqm before merging?

**Comment by steveperry-53** [comment]
Added Tech Review LGTM based on lgtm by vish. Might want to wait a bit for @tengqm to chime in.
@zacharysarah, I'll leave that up to you.

**Comment by zacharysarah** [comment]
@steveperry-53 Agreed, let's give @tengqm a chance to respond. 👍

**Comment by ConnorDoyle** [comment]
@tengqm thanks! let's land it 👍

**Pull request merged**